src/series/ops/arithmetic.rs

```diff
@@ -14,6 +14,16 @@ macro_rules! impl_series_math_op {
             type Output = DaftResult<Series>;
             fn $func_name(self, rhs: Self) -> Self::Output {
                 let (lhs, rhs) = match_types_on_series(self, rhs)?;
+                if !lhs.data_type().is_numeric() {
```

Comment:
can make this a or since the both the bodies are the same

Reply:
Ah good catch, I actually wanted to have the error say `lhs/rhs`. But I'll combine them as you suggested.